### PR TITLE
RC 801: Feature enhance the Sunbird RC CLI

### PR DESCRIPTION
**Pull request by Sreejit-K**
Screenshot from 2023-09-19 17-43-15

Registry-CLI updates : 

1. Provides the options to enable/ disable - 
    -  Authentication in registry
    -  Elastic search
    -  Attestation workflows 
    -  Definitions Management 
    -  Synchronous or asynchronous generation of entities 
    -  Auxiliary services ( Notification ms, Metrics ms, context-proxy-service, Digi-locker and Bulk Issuance. )
   
2. Generates public/private keys, that are required for Registry setup and signing.
3. We have observed that the PC hangs for most of users while using the CLI due to the lack of system configurations to run all the services provided by the registry core. But the updates solves this problem by providing the users with the flexibility to enable and disable services based on their use cases. And on using the default configurations after the updates. It runs the basic services with are needed to start with registry i.e. DB, Keycloak and Registry.

### CODE REVIEW

File: tools/cli/pnpm-lock.yaml [hunk 1]
```diff
@@ -22,6 +23,7 @@ specifiers:
 dependencies:
   '@types/node': 16.11.21
   '@types/update-notifier': 5.1.0
+  apisauce: 2.1.5
```
[the remark this answers is not in the document]
Comment (Sreejit-K) [reply tools/cli/pnpm-lock.yaml:26]: This is used to call the health check  and Key cloak API's. This was already being used in the code. But not sure how it got removed from the dependencies. Have add is as the build & install was failing.

File: tools/cli/contributing.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ v16.13.0
 Once NodeJS is installed, run the following in terminal to install PNPM:
 
 ```sh
-$ sudo npm install --global pnpm
+$ sudo npm install --global pnpm@6.20.1
```
[the remark this answers is not in the document]
Comment (Sreejit-K) [reply tools/cli/contributing.md]: we don't have to use a specific version. Just used the version mentioned in the contributing doc as I was facing an issue while build/ install. But resolved it, by installing the api-sauce dependency. Will revert this. Thanks for pointing out.

File: tools/cli/package.json [hunk 3]
```diff
@@ -61,7 +70,7 @@
 	},
 	"pnpm": {
 		"overrides": {
-			"ansi-regex@>2.1.1 <5.0.1": ">=5.0.1",
+			"ansi-regex@>3.0.1 <5.0.1": ">=5.0.1",
```
[the remark this answers is not in the document]
Comment (Sreejit-K) [reply tools/cli/package.json:73]: As I was using a older version of the pnpm, it was throwing a few ansi-regex deprecated functions errors.  Have reverted it and and tested it working fine with the latest version of pnpm.

File: docker-compose.yml [hunk 4]
```diff
@@ -293,29 +319,36 @@ services:
...
     image: ghcr.io/sunbird-rc/sunbird-rc-digilocker-certificate-api
     volumes:
       - ./services/digilocker-certificate-api/config/:/go/config/
     ports:
-      - "8087:8087"
+      - '8087:8087'
     environment:
       KEYCLOAK_CLIENT_ID: admin-api
-      KEYCLOAK_CLIENT_SECRET: ${KEYCLOAK_SECRET}
+      KEYCLOAK_CLIENT_SECRET: 61fdeaf0-9654-4c12-b7ff-f7116c9e4fca
```
Comment (holashchand) [inline docker-compose.yml:331]: why did you put secret here? plz revert

**Review by holashchand — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 4 above)
Comment (Sreejit-K) [reply docker-compose.yml]: Reverted! This change was made while testing the CLI.

**Review by holashchand — APPROVED** [review]

**Review by srprasanna — APPROVED** [review]

**Pull request merged**